src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1144,6 +1166,120 @@ export default class SourceFileLinter {
 		}
 	}
 
+	#analyzeNewAndApplySettings(node: ts.NewExpression | ts.CallExpression) {
+		node?.arguments?.filter((arg) => ts.isObjectLiteralExpression(arg))
+			.flatMap((arg) => arg.properties)
+			.forEach((prop) => {
+				if (ts.isPropertyAssignment(prop) &&
+					(ts.isCallExpression(node) ||
+						this.#isPropertyBindingType(node, getPropertyNameText(prop.name)))
+				) {
+					if (ts.isObjectLiteralExpression(prop.initializer)) {
+						this.#analyzeModelDataTypes(prop.initializer);
+					} else {
+						this.#analyzeModelDataTypesBinding(prop);
+					}
+				}
+			});
+	}
+
+	#analyzeModelDataTypes(node: ts.ObjectLiteralExpression) {
+		node?.properties.forEach((prop) => {
+			if (!ts.isPropertyAssignment(prop)) {
+				return;
+			}
+
+			// Get the type property
+			let typeField;
+			if (ts.isObjectLiteralExpression(prop.initializer)) {
+				typeField = prop.initializer.properties.find((prop: ts.ObjectLiteralElementLike) => {
+					return ts.isPropertyAssignment(prop) &&
+						ts.isIdentifier(prop.name) &&
+						prop.name.text === "type";
+				});
```

Comment:
Can be replaced with `getPropertyAssignmentInObjectLiteralExpression` from the utils.